### PR TITLE
[FLINK-28271][runtime] Add config options for speculative scheduler

### PR DESCRIPTION
**Pull request by zhuzhurk**
## What is the purpose of the change

This PR adds config options for speculative scheduler. 
It is needed to unblock the development of blocklist and speculative scheduler.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by zhuzhurk** [comment]
@flinkbot run azure

File: flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java [hunk 1]
```diff
@@ -593,6 +593,40 @@ public enum SchedulerType {
                                             code(SchedulerType.AdaptiveBatch.name()))
                                     .build());
 
+    @Documentation.Section({
+        Documentation.Sections.EXPERT_SCHEDULING,
+        Documentation.Sections.ALL_JOB_MANAGER
+    })
+    public static final ConfigOption<Boolean> SPECULATIVE_ENABLED =
```
Comment (wanglijie95) [inline flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java:600]: I perfer to add a `ADAPTIVE_BATCH_SCHEDULER` prefix, the same goes for the other two options.

**Review by wanglijie95 — APPROVED** [review]
Thanks for creating this PR @zhuzhurk. 
LGTM, I left 2 minor comment, please take a look.

File: flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java (hunk 1 above)
Comment (zhuzhurk) [reply flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java:600]: I prefer to keep the option name not too long, as long as the we can identify it is part of speculative execution.
I will rename `BLOCK_SLOW_NODE_DURATION` to `SPECULATIVE_BLOCK_SLOW_NODE_DURATION`.

**Comment by zhuzhurk** [comment]
@flinkbot run azure

**Comment by zhuzhurk** [comment]
@flinkbot run azure

**Pull request closed without merging**